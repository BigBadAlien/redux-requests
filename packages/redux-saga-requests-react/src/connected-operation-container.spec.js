import renderer from 'react-test-renderer';
import React from 'react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

import ConnectedOperationContainer from './connected-operation-container';

const mockStore = configureStore();

describe('ConnectedOperationContainer', () => {
  it('maps requestSelector to operation with operationType', () => {
    const operationType = 'OPERATION';
    const component = renderer.create(
      <Provider
        store={mockStore({
          request: {
            operations: { [operationType]: { pending: 1, error: 'error' } },
          },
        })}
      >
        <ConnectedOperationContainer
          requestSelector={state => state.request}
          operationType={operationType}
        >
          {({ loading, error }) => (
            <div>
              {loading && 'loading'}
              {error}
            </div>
          )}
        </ConnectedOperationContainer>
      </Provider>,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('maps requestSelector to operation with operationActionCreator and passes sendOperation', () => {
    const operationType = 'OPERATION';
    const operationCreator = () => ({
      type: operationType,
      request: { url: '/' },
    });
    operationCreator.toString = () => operationType;
    const component = renderer.create(
      <Provider
        store={mockStore({
          request: {
            operations: { [operationType]: { pending: 1, error: 'error' } },
          },
        })}
      >
        <ConnectedOperationContainer
          requestSelector={state => state.request}
          operationCreator={operationCreator}
        >
          {({ loading, error, sendOperation }) => (
            <div>
              {loading && 'loading'}
              {error}
              <button type="button" onClick={sendOperation}>
                Send operation
              </button>
            </div>
          )}
        </ConnectedOperationContainer>
      </Provider>,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});
