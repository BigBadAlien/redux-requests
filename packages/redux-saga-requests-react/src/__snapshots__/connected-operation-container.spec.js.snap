// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConnectedOperationContainer maps requestSelector to operation with operationActionCreator and passes sendOperation 1`] = `
<div>
  loading
  error
  <button
    onClick={[Function]}
    type="button"
  >
    Send operation
  </button>
</div>
`;

exports[`ConnectedOperationContainer maps requestSelector to operation with operationType 1`] = `
<div>
  loading
  error
</div>
`;
